dev/PushNotifications/PushNotificationsLongRunningTask/platform.cpp

```diff
@@ -45,58 +46,60 @@ void NotificationsLongRunningPlatformImpl::WaitForLifetimeEvent()
     m_lifetimeManager.Wait();
 }
 
-void NotificationsLongRunningPlatformImpl::AddToRegistry(const std::wstring& processName, std::wstring appId)
+wil::unique_cotaskmem_string NotificationsLongRunningPlatformImpl::GetAppIdentifier(const std::wstring& processName)
 {
-    std::wstring subkey = L"Software\\Microsoft\\Windows\\CurrentVersion\\PushNotifications\\LongRunningProcess";
-
-    wil::unique_hkey hKeyResult;
-
-    THROW_IF_WIN32_ERROR(RegCreateKeyEx(
-        HKEY_CURRENT_USER,
-        subkey.c_str(),
-        0,
-        nullptr,
-        REG_OPTION_NON_VOLATILE,
-        KEY_ALL_ACCESS,
-        nullptr,
-        &hKeyResult,
-        nullptr));
-
-    THROW_IF_WIN32_ERROR(RegSetKeyValue(
-        hKeyResult.get(),
-        nullptr,
-        appId.c_str(),
-        REG_SZ,
-        processName.c_str(),
-        static_cast<DWORD>(sizeof(WCHAR) * processName.size())));
-}
+    auto values{ m_storage.Values() };
 
-void NotificationsLongRunningPlatformImpl::GetAppIdentifier(const std::wstring processName)
-{
-    if (m_appIdMap.find(processName) == m_appIdMap.end())
+    for (auto it = values.begin(); it != values.end(); it++)
     {
-        GUID guidReference;
-        THROW_IF_FAILED(CoCreateGuid(&guidReference));
-
-        wil::unique_cotaskmem_string guidStr;
-        THROW_IF_FAILED(StringFromCLSID(guidReference, &guidStr));
+        winrt::hstring settingValue{ winrt::unbox_value<winrt::hstring>(it.Current().Value()) };
 
-        AddToRegistry(processName, guidStr.get());
-        m_appIdMap[processName] = guidStr.get();
+        if (processName.compare(settingValue.c_str()) == 0)
+        {
+            return wil::make_unique_string<wil::unique_cotaskmem_string>(it.Current().Key().c_str());
+        }
     }
+
+    GUID guidReference;
+    THROW_IF_FAILED(CoCreateGuid(&guidReference));
+
+    wil::unique_cotaskmem_string guidStr;
+    THROW_IF_FAILED(StringFromCLSID(guidReference, &guidStr));
+
+    m_storage.Values().Insert(guidStr.get(), winrt::box_value(processName.c_str()));
```

Comment:
nit: can this just be `values.Insert(guidStr.get(), winrt::box_value(processName.c_str()));`

No need to address though.

Reply:
Might take this in the big PR.